### ISSUE TITLE
iPadOS: Ctrl + C in terminal doesn’t kill running process

### ISSUE THREAD

**Issue by jsjoeio** [microsoft/vscode#114009]
## Description 

Hitting `ctrl` + `c` with the terminal focused and a process running (i.e. `yarn start` in a React app) does not kill the process. 

If someone is able to provide a good starting point, I would be willing to take a stab at this.

- VSCode Version: 1.53.0-insider
- OS Version: iPadOS 14.1 (18A8395)
- Commit: 01089c0a5052432594a177f90ad9f3148ec2a6a2
- Date: 2021-01-07T05:25:51.740Z (16 hrs ago)
- Browser: Mozilla/5.0 (iPad; CPU OS 14_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) CriOS/87.0.4280.77 Mobile/15E148 Safari/604.1

## Steps to Reproduce:

1. On iPad, open Codespaces environment in Chrome
2. Open integrated Terminal
3. Start application (i.e. like a create-react-app - `yarn start`)
4. Focus terminal and hit `ctrl` + `c`

## Expected

Process (`yarn start`) should stop running

## Actual
Terminal receives new line

https://user-images.githubusercontent.com/3806031/103948200-a11ad800-50f6-11eb-8cb4-87e951735911.MP4

Does this issue occur when all extensions are disabled?: Yes

Related:
- https://github.com/cdr/code-server/issues/1343

**Comment by luca400** [comment]
Perhaps that is not the right sigint.

**Comment by jsjoeio** [comment]
One other piece of insight: this may not be limited to VSCode. @cdr has a product which is a terminal in the browser. I’m able to reproduce the same issue (see line breaks at the bottom).

![9949BCB2-5A05-442E-987A-A2DF4D297061](https://user-images.githubusercontent.com/3806031/104030342-8002c780-5188-11eb-8327-2daee89be615.png)

**Comment by jsjoeio** [comment]
As a temporary workaround, you can add a shortcut like so:
1. Open Command Palette 
2. Look for "Preferences: Open Keyboard Shortcuts (JSON)"
3. Add this:
```json
{
    "key": "ctrl+shift+c",
    "command": "workbench.action.terminal.sendSequence",
    "args": {
      "text": "\u0003"
    },
    "when": "terminalFocus"
}
```
Source: [StackOverflow](https://stackoverflow.com/a/52735954/3015595)

**Comment by springhack** [comment]
I found the root reason was iPad Safari report keyCode of ctrl-c as 13, which is not correct, reference from MDN, keyCode is a deprecated property;

The key and code property is right, so I suggest just use key property, not keyCode, on the whole html page not just terminal.

**Comment by rebornix** [comment]
@meganrogge I'm still seeing this in codespace, am I testing the right environment?

**Comment by meganrogge** [comment]
fixed w cc077af2d6fd1ef8c06c4dbc759373f412c21d8e

**Comment by lxjwlt** [comment]
as another workaround, use `ctrl+\` shortcut instead of `ctrl+c`, it will interrupt the process as well
